src/components/KPop/KPop.vue

```diff
@@ -1,431 +1,306 @@
-    }
+  title: {
+    type: String,
+    default: '',
   },
-  computed: {
-    popoverStyle: function() {
-      return {
-        width: getSizeFromString(this.width),
-        maxWidth: getSizeFromString(this.maxWidth),
-        maxHeight: getSizeFromString(this.maxHeight),
-      }
-    },
-    popoverClassObj: function() {
-      return [this.popoverClasses, { 'hide-caret': this.hideCaret }]
-    },
+  placement: {
+    type: String as PropType<PopPlacements>,
+    validator: (value: PopPlacements): boolean => PopPlacementsArray.includes(value),
+    default: 'auto',
   },
-  watch: {
-    hidePopover: function() {
-      // whenever this prop gets updated, hide the popper
-      if (this.isOpen) {
-        this.hidePopper()
-      }
-    },
-    disabled: {
-      handler() {
-        if (this.isOpen) {
-          this.hidePopper()
-        }
-      },
-      immediate: true,
-    },
+  trigger: {
+    type: String as PropType<PopTrigger>,
+    default: 'click',
+    validator: (value: PopTrigger): boolean => PopTriggerArray.includes(value),
   },
-  mounted() {
-    if (!this.$el.children) {
-      this.reference = this.$el
-    } else {
-      this.reference = this.$el.children[0]
-    }
-    this.bindEvents()
+  popoverTimeout: {
+    type: Number,
+    default: 300,
+  },
+  hideCloseIcon: {
+    type: Boolean,
+    default: false,
+  },
+  hideCaret: {
+    type: Boolean,
+    default: false,
+  },
+  closeOnPopoverClick: {
+    type: Boolean,
+    default: false,
+  },
+  disabled: {
+    type: Boolean,
+    default: false,
+  },
+  width: {
+    type: String,
+    default: '200',
+  },
+  maxWidth: {
+    type: String,
+    default: 'auto',
+  },
+  maxHeight: {
+    type: String,
+    default: 'auto',
+  },
+  popoverClasses: {
+    type: String,
+    default: '',
+  },
+  tag: {
+    type: String,
+    default: 'div',
+  },
+  zIndex: {
+    type: Number,
+    default: 1000,
   },
-  beforeUnmount() {
-    const popper = this.$refs.popper
+  positionFixed: {
+    type: Boolean,
+    default: true,
+  },
+})
 
-    if (document) {
-      document.documentElement.removeEventListener('click', this.handleClick)
-    }
+const emit = defineEmits(['open', 'close', 'popover-click'])
 
-    if (popper) {
-      popper.removeEventListener('click', this.onPopperContentClick)
-      popper.removeEventListener('mouseenter', this.showPopper)
-      popper.removeEventListener('focusin', this.showPopper)
-      popper.removeEventListener('mouseleave', this.hidePopper)
-      popper.removeEventListener('focusout', this.hidePopper)
-    }
+const { getSizeFromString } = useUtilities()
 
-    if (this.reference) {
-      this.reference.removeEventListener('mouseenter', this.createInstance)
-      this.reference.removeEventListener('mouseleave', this.toggle)
-      this.reference.removeEventListener('focus', this.createInstance)
-      this.reference.removeEventListener('blur', this.toggle)
+const popoverId = uuid4()
+const titleId = uuid4()
```

Comment:
question: remind me why we need these ids?

Reply:
`aria-labelledby`, `aria-controls`